File: clojure/nth_prime.clj
(ns nth-prime
  (:require [clojure.math :as math]))

(defn- prime?
  [n]
  (cond
    (< n 2) false
    (= n 2) true
    :else
    (let [sq (->> n (math/sqrt) (math/round) (inc) (range 2))]
      (every? #(> (rem n %) 0) sq))))

(defn nth-prime
  [n]
  (if (< n 1)
    (throw (IllegalArgumentException. "there is no zeroth prime"))
    (loop [n n p 2]
      (if (prime? p)
        (if (= n 1)
          p
          (recur (dec n) (inc p)))
        (recur n (inc p))))))

File: python/nth_prime.py
def is_prime(n: int) -> bool:
    if n < 2:
        return False
    if n == 2:
        return True

    m = round(n**0.5) + 1
    for i in range(2, m):
        if n % i == 0:
            return False
    return True


def prime(number: int) -> int:
    if number == 0:
        raise ValueError('there is no zeroth prime')
    n = 1
    while number > 0:
        n += 1
        if is_prime(n):
            number -= 1
    return n
